addons/portal/wizard/mail_invite_share.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import werkzeug.urls
+
+from odoo import api, fields, models, _
+
+
+class MailInviteShare(models.TransientModel):
+    _name = 'mail.invite.share'
+
+    def default_share_link(self):
+        active_model = self.env.context.get('active_model')
+        active_id = self.env.context.get('active_id')
+        res = False
+        if active_model:
+            model = self.env[active_model]
+            if isinstance(model, self.pool['portal.mixin']):
+                doc_url = model.browse(active_id).portal_url
+                base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+                res = base_url + doc_url
+        return res
+
+    share_link = fields.Char(string="Document link", default=default_share_link)
+    partner_ids = fields.Many2many('res.partner', string="Recipients", required=True)
+    note = fields.Text(help="Add extra content that user want to display in sent mail")
```

Comment:
help message would be clearer with something like this : "Add extra content to display in the email"